store/src/sql/event.rs

```diff
@@ -338,6 +338,14 @@ impl SqliteEventStore {
             .await?;
         Ok(())
     }
+
+    async fn get_block(&self, cid: &Cid) -> Result<Option<Vec<u8>>> {
+        let block: Option<BlockBytes> = sqlx::query_as(BlockQuery::get())
+            .bind(cid.hash().to_bytes())
+            .fetch_optional(self.pool.reader())
```

Comment:
why is this `fetch_optional` when everything else in this file uses `fetch_all`?  I don't actually know what either of those functions do, so mostly just looking to increase my own understanding of how this all works.

Reply:
fetch_all returns a Vec,T> that may be empty of no data exists for the query. fetch_optional returns Option<T>. If fetch_optional find more than one value an error is returned. If we use fetch_all we have to explicitly check  that the vector.len <= 1.  In this case we expect at most one value to match the query so fetch_optional is the most appropriate method to use.